source/extensions/original_ip_detection/custom_header/custom_header.cc

```diff
@@ -0,0 +1,50 @@
+#include "extensions/original_ip_detection/custom_header/custom_header.h"
+
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace OriginalIPDetection {
+namespace CustomHeader {
+
+CustomHeaderIPDetection::CustomHeaderIPDetection(
+    const envoy::extensions::original_ip_detection::custom_header::v3::CustomHeaderConfig& config)
+    : header_name_(config.header_name()),
+      allow_trusted_address_checks_(config.allow_extension_to_set_address_as_trusted()) {
+
+  if (config.has_reject_options()) {
+    Http::Code reject_code = Http::Code::Forbidden;
+
+    if (config.reject_options().has_status_on_error()) {
+      reject_code = static_cast<Http::Code>(config.reject_options().status_on_error().code());
```

Comment:
Hm, do you think we should validate the code is a reasonable response code on config ingress?

Reply:
where reasonable is >= 400 <= 599?

Reply:
added